src/compiler/types.ts

```diff
@@ -6458,6 +6458,8 @@ namespace ts {
         primary: boolean;
         // The location of the .d.ts file we located, or undefined if resolution failed
         resolvedFileName: string | undefined;
+        /** The location of the symlink to the .d.ts file we found, if `resolvedFileName` was the realpath. */
+        originalFileName: string | undefined;

```

Comment:
You also want to make this optional esp given this is API...  might break if someone is setting these things manually..